Cesium3DTiles/src/Tileset.cpp

```diff
@@ -1143,6 +1147,127 @@ static void markTileAndChildrenNonRendered(
   markChildrenNonRendered(lastFrameNumber, lastResult, tile, result);
 }
 
+/**
+ * @brief Returns whether the tile is completely clipped out by a clipping
+ * polygon.
+ *
+ * @param boundingVolume The bounding volume of the tile.
+ * @param cartographicSelections The list of clipping selections to check.
+ * @return Whether the tile is completely clipped out by a clipping polygon.
+ */
+static bool isCompletelyClipped(
+    const BoundingVolume& boundingVolume,
+    const std::vector<CartographicSelection>& cartographicSelections) {
```

Comment:
This function is expensive to run for every tile every frame, especially with a complicated polygon. We should be able to optimize it by adding quick accept/reject checks. For example, if a tile's bounding region is entirely disjoint with the polygon's bounding region, there's no need to check every triangle in the polygon.